### PR TITLE
arangodb 3.5.1

### PR DESCRIPTION
**Pull request by KVS85**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by fxcoudert** [comment]
On Catalina the test fails with:

```
==> /usr/local/Cellar/arangodb/3.5.1/bin/arangosh --server.password "" --javascript.execute-string "require('@arangodb').print('it works!')"
Error: arangodb: failed
An exception occurred within a child process:
  Test::Unit::AssertionFailedError: <0> expected but was
<nil>.
```

**Comment by SMillerDev** [comment]
@brewtestbot test this please!

File: Formula/arangodb.rb [hunk 1]
```diff
@@ -28,8 +28,11 @@ class Arangodb < Formula
       :revision => "bbe29730e70dba609b57c469e8f863f032fabf3e"
   end
 
+  patch :DATA
```
Comment (fxcoudert) [inline Formula/arangodb.rb]: Please make sure the bug is reported upstream, and add a link in a comment above

File: Formula/arangodb.rb [hunk 2]
```diff
@@ -28,8 +28,11 @@ class Arangodb < Formula
...
+
   def install
-    ENV.cxx11
+    ENV.cxx
```
Comment (fxcoudert) [inline Formula/arangodb.rb]: `ENV.cxx` is the name of the C++ compiler. This line should simply be removed.

File: Formula/arangodb.rb [hunk 3]
```diff
@@ -45,26 +48,28 @@ def install
...
+        -DUSE_CATCH_TESTS=Off
+        -DUSE_GOOGLE_TESTS=Off
+        -DCMAKE_INSTALL_LOCALSTATEDIR=#{var}
       ]
 
       if ENV.compiler == "gcc-6"
         ENV.append "V8_CXXFLAGS", "-O3 -g -fno-delete-null-pointer-checks"
       end
 
+      ENV.append "CXXFLAGS", "-std=c++14"
```
Comment (fxcoudert) [inline Formula/arangodb.rb]: Why is this not detected by the build system?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/arangodb.rb (hunk 2 above)
Comment (fceller) [reply Formula/arangodb.rb]: line removed

File: Formula/arangodb.rb (hunk 1 above)
Comment (fceller) [reply Formula/arangodb.rb]: added link to upstream patch

File: Formula/arangodb.rb (hunk 3 above)
Comment (fceller) [reply Formula/arangodb.rb]: removed. Maybe this was necessary before I removed the link ENV.cxx11

File: Formula/arangodb.rb [hunk 4]
```diff
@@ -28,8 +28,11 @@ class Arangodb < Formula
       :revision => "bbe29730e70dba609b57c469e8f863f032fabf3e"
   end
 
+  # fixed upstream in https://github.com/arangodb/arangodb/commit/55ca6c8660e0b373bae1840d61894a4202913cfa
+  patch :DATA
```
Comment (fxcoudert) [inline Formula/arangodb.rb]:
```suggestion
  patch do
    url "https://github.com/arangodb/arangodb/commit/55ca6c86.diff?full_index=1"
    sha256 "99f653d868360949ee6eb2d5fa380ded3f9a8941caba1ce7b97c88b6e4ae166d"
  end
```

Comment (fxcoudert) [reply Formula/arangodb.rb]: This is easier to maintain than having the patch added to the formula itself.

**Review by fxcoudert — DISMISSED** [review]

File: Formula/arangodb.rb (hunk 4 above)
Comment (KVS85) [reply Formula/arangodb.rb]: Added. Thanks for mention!

**Comment by KVS85** [comment]
@fxcoudert can you please merge it?

**Comment by KVS85** [comment]
@SMillerDev can you please merge it?

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
@BrewTestBot test this please

**Comment by MikeMcQuaid** [comment]
Thanks so much for your contribution! Without people like you submitting PRs we couldn't run this project. You rock, @KVS85!

**Pull request closed without merging**